sampleExtension/background.js

```diff
@@ -1,60 +1,117 @@
-    updateBlockedSites(replaceListener);
-    addBrowserActionListener();
+/* --------------- ------ update list of BlockedSites ------ ---------------*/
+
+updateStorage = function() {
+    setStorageBlacklist(blockedSites);
+    return {redirectUrl: "https://zeeguu.herokuapp.com/get-ex"};
 };
 
 // This function receives the blacklist from the sync storage.
-updateBlockedSites = function(callback){
-    chrome.storage.sync.get("tds_blacklist", function (items) {
-        if (!chrome.runtime.error) {
-            blockedSites = items.tds_blacklist;
-            return callback();
-        }
+retrieveBlockedSites = function(callback){
+    getStorageBlacklist(function(blacklist) {
+        blockedSites = blacklist;
+        return callback();
+    });
+};
+
+//Loads the intercept time+date list from storage
+retrieveInterceptDateList = function() {
+    getInterceptDateList(function(dateList) {
+        interceptDateList = dateList;
+    });
+};
+
+//Loads the intercept counter from storage
+retrieveInterceptCounter = function(callback) {
+    getInterceptCounter(function(counter) {
+        interceptCounter = counter;
+        return callback();
     });
 };
 
 // This function adds one url to the blacklist
 addToBlockedSites = function(urlToAdd) {
-    var formatUrl = urlToAdd.split(['//'])[1];
-    formatUrl = '*://' + formatUrl + '*';
-    blockedSites.push(formatUrl);
-    chrome.storage.sync.set({"tds_blacklist" : blockedSites }, function() {
-        if(chrome.runtime.error) {
-            console.log("Runtime error.");
-        }
-    });
+    var blockedSiteItem = new BlockedSite(formatUrl(urlToAdd));
+    blockedSites.push(blockedSiteItem);
+    updateStorage();
+    replaceListener();
+};
+
+// This function adds the current time+date to the saved time+date list
+addToInterceptDateList = function() {
+    var newDate = new Date().toDateString();
+    if (interceptDateList == null) {
+        interceptDateList = [newDate];
+    } else {
+        interceptDateList.push(newDate);
+    }
+    setInterceptDateList(interceptDateList);
+};
+
+//TODO fix in interation 3 for url-handling module
+formatUrl = function(url) {
+    result = url.split(['//'])[1];
+    result = result.split("").reverse().join("");
+    result = result.split(['/'])[1];
+    result = result.split("").reverse().join("");
+    return result;
 };
 
 /* --------------- ------ Listener functions ------ ---------------*/
 
 replaceListener = function() {
-    chrome.webRequest.onBeforeRequest.removeListener(intercept);
-    addWebRequestListener();
+    chrome.webRequest.onBeforeRequest.removeListener(handleInterception);
+    urlList = filterBlockedSitesOnChecked();
+    if (urlList.length > 0) {
+        addWebRequestListener(urlList);
+    }
+};
+
+filterBlockedSitesOnChecked = function() {
+    if(blockedSites.length > 0) {
+        return blockedSites.filter(function (a) {return a.checkboxVal == true;});
```

Comment:
how does this have too many levels of abstraction? could you please explain. It is one function, that filters a list.
The filter function is defined this way, you need to pass it a boolean function which it can apply to every element. This is the way how filter-functions are implemented almost everywhere.  So please explain how we should change this so that it does not have "too many levels of abstraction". Or just a general explanation of this term that is used constantly everywhere :)